--- lib_initscr.c	Sat Mar  8 15:03:58 1997
+++ /usr/src/ncurses-4.1/ncurses/lib_initscr.c	Sat Jun  7 15:04:14 1997
@@ -75,7 +75,7 @@
 		initialized = TRUE;
 
 		if ((name = getenv("TERM")) == 0)
-			name = "unknown";
+			name = "vt52";
 		if (newterm(name, stdout, stdin) == 0) {
 			fprintf(stderr, "Error opening terminal: %s.\n", name);
 			exit(EXIT_FAILURE);
@@ -102,5 +102,7 @@
 		(void) strncpy(ret, term, sizeof(ret) - 1);
 		term = ret;
 	}
+	else
+		strcpy(ret, "vt52");
 	return term;
 }
