diff --git a/deps/ncurses-4.1/ncurses/lib_getch.c b/deps/ncurses-4.1/ncurses/lib_getch.c
index d59531f..742f141 100644
--- a/deps/ncurses-4.1/ncurses/lib_getch.c
+++ b/deps/ncurses-4.1/ncurses/lib_getch.c
@@ -328,6 +328,8 @@ int	ch;
 **
 */
 
+extern int __mint;
+
 static int
 kgetch(WINDOW *win GCC_UNUSED)
 {
@@ -367,7 +369,7 @@ int timeleft = ESCDELAY;
 					break;
 
 				TR(TRACE_IEVENT, ("waiting for rest of sequence"));
-				if (!_nc_timed_wait(3, timeleft, &timeleft)) {
+				if (__mint && !_nc_timed_wait(3, timeleft, &timeleft)) {
 					TR(TRACE_IEVENT, ("ran out of time"));
 					return(fifo_pull());
 				} else {
