--- lib_setup.c	Sat Mar  8 22:25:44 1997
+++ /usr/src/ncurses-4.1/ncurses/lib_setup.c	Sat Jun  7 17:37:24 1997
@@ -206,7 +206,8 @@
 	if (tname == NULL) {
 		tname = getenv("TERM");
 		if (tname == NULL)
-			ret_error0(-1, "TERM environment variable not set.\n");
+/*			ret_error0(-1, "TERM environment variable not set.\n");*/
+			tname = "vt52";
 	}
 
 	T(("your terminal name is %s", tname));
@@ -257,7 +258,8 @@
 	 */
 	if (Filedes == STDOUT_FILENO && !isatty(Filedes))
 	    Filedes = STDERR_FILENO;
-	cur_term->Filedes = Filedes;
+/*	cur_term->Filedes = Filedes; with Filedes = 1 read() doesn't work :-( */
+	cur_term->Filedes = 0;	/* with Filedes = 0 everything works as intended */
 
 	_nc_get_screensize();
 
